src/testing/cluster/storage_checker.zig

```diff
@@ -195,26 +194,44 @@
         }
     }
 
-    fn checksum_trailer(superblock: *const SuperBlock, trailer: vsr.SuperBlockTrailer) u128 {
-        const trailer_size = superblock.working.trailer_size(trailer);
-        const trailer_checksum = superblock.working.trailer_checksum(trailer);
+    fn checksum_client_replies(checker: *StorageChecker, superblock: *const SuperBlock) u128 {
+        assert(superblock.working.vsr_state.sync_op_max == 0);
 
-        var copy: u8 = 0;
-        while (copy < constants.superblock_copies) : (copy += 1) {
-            const trailer_start = trailer.zone().start_for_copy(copy);
+        const client_sessions_size = superblock.working.vsr_state.checkpoint.client_sessions_size;
+        if (client_sessions_size > 0) {
+            const checkpoint = &superblock.working.vsr_state.checkpoint;
+            var client_sessions_block: vsr.BlockReference = .{
+                .address = checkpoint.client_sessions_last_block_address,
+                .checksum = checkpoint.client_sessions_last_block_checksum,
+            };
 
-            assert(trailer_checksum ==
-                vsr.checksum(superblock.storage.memory[trailer_start..][0..trailer_size]));
-        }
+            var client_sessions_cursor: usize = client_sessions_size;
+            while (true) {
+                const block =
+                    superblock.storage.grid_block(client_sessions_block.address).?;
+                assert(schema.header_from_block(block).checksum == client_sessions_block.checksum);
 
-        return trailer_checksum;
-    }
+                const block_body = schema.TrailerNode.body(block);
+                client_sessions_cursor -= block_body.len;
+                stdx.copy_disjoint(
+                    .inexact,
+                    u8,
+                    checker.client_sessions_buffer[client_sessions_cursor..],
+                    block_body,
+                );
 
-    fn checksum_client_replies(superblock: *const SuperBlock) u128 {
-        assert(superblock.working.vsr_state.sync_op_max == 0);
+                client_sessions_block = schema.TrailerNode.previous(block) orelse break;
+            }
+            assert(client_sessions_cursor == 0);
+        }
+        assert(vsr.checksum(checker.client_sessions_buffer[0..client_sessions_size]) ==
+            superblock.working.vsr_state.checkpoint.client_sessions_checksum);
+
+        checker.client_sessions.decode(checker.client_sessions_buffer[0..client_sessions_size]);
+        defer checker.client_sessions.reset();
 
         var checksum: u128 = 0;
```

Comment:
Maybe use steraming checksum instead of `^` here to add dependency on the _order_ of replies and
not just their content?